client/blocks/payment-request/index.js

```diff
@@ -0,0 +1,76 @@
+/**
+ * External dependencies
+ */
+import { getSetting } from '@woocommerce/settings';
+
+/**
+ * Internal dependencies
+ */
+import { PAYMENT_METHOD_NAME } from './constants';
+import { PaymentRequestExpress } from './payment-request-express';
+import { applePayImage } from './apple-pay-preview';
+import { getStripeServerData, loadStripe } from '../stripe-utils';
+
+const ApplePayPreview = () => <img src={ applePayImage } alt="" />;
+
+const canPayStripePromise = loadStripe();
+const componentStripePromise = loadStripe();
+
+let isStripeInitialized = false,
+	canPay = false;
+
+// Initialise stripe API client and determine if payment method can be used
+// in current environment (e.g. geo + shopper has payment settings configured).
+function paymentRequestAvailable( { currencyCode, totalPrice } ) {
+	// Stripe only supports carts of greater value than 30 cents.
+	if ( totalPrice < 30 ) {
+		return false;
+	}
```

Comment:
It's actually 50 US cents. But I wouldn't add this type of logic in here, because it depends on currency and Stripe's rules.

On top of that, this isn't something that's verified in the shortcode or the credit card implementation.

Reply:
I'll remove it 👍

Reply:
For reference, this was [originally added in the blocks integration here](https://github.com/woocommerce/woocommerce-gutenberg-products-block/pull/3424)

Reply:
Removed in 5e8569be7d2a84f9cbad59928fb0cbed24780f51

Reply:
@reykjalin - what happens when the total is less than the minimum set by Stripe?

Reply:
We just display an error from Stripe:

### Shortcode version

(PRB = Payment Request button)

| Product + PRB | Cart + PRB | Checkout + Card |
| :---: | :---: | :---: |
| ![image](https://user-images.githubusercontent.com/13835680/118270396-8a72fc00-b4af-11eb-84ba-a4509ecadc47.png) | ![image](https://user-images.githubusercontent.com/13835680/118271189-7bd91480-b4b0-11eb-8a72-5d8f9b453e79.png) | ![image](https://user-images.githubusercontent.com/13835680/118271388-b478ee00-b4b0-11eb-87c6-7960f26531f9.png) |

### Blocks version

| Cart + PRB | Checkout + PRB | Checkout + Card |
| :---: | :---: | :---: |
| ![image](https://user-images.githubusercontent.com/13835680/118270541-baba9a80-b4af-11eb-9254-0367297c9daf.png) | ![image](https://user-images.githubusercontent.com/13835680/118270663-df167700-b4af-11eb-9d58-cbe5687150ea.png) | ![image](https://user-images.githubusercontent.com/13835680/118270773-01a89000-b4b0-11eb-9d62-e2cc731361c4.png) |

Reply:
Just noting this is why in the integration for blocks we intentionally did not show the payment method(s) if the minimum order total wasn't supported (poor UX for the user, especially if there are other ways to check out). However, in hindsight I could see some counters to our decision there:

- The payment gateway could change what they support for minimum total.
- One could argue that showing this kind of error is actually beneficial because it's is incentive for the shopper to add something else to their order, or increase quantity (however I'd posit that there's also greater risk of abandonment at this point because there's no clear action for the shopper).
- This is more of a side-effect of the merchant not realizing that the gateway's have minimum requirements and so they should be setting their prices accordingly (so more of an edgecase).  By having this notice, merchants should be able to catch this in testing (in theory).

It's an example of where with the new checkout flow, we took the opportunity to evaluate whether existing behaviour in the shortcode checkout flow had behaviour worth preserving in the new checkout flow or if it was a good opportunity to improve.

Anyways, I just thought I'd share some context around why that logic was added.

Reply:
I do think it's best to rely on the response from Stripe because they can change the value any time they see fit. The other reasons you outlined are also a factor, but mostly I don't think we should be enforcing something we have no control over 🤔 

Thanks for chiming in Darren! It's great to get your perspective on this ☺️